htlcswitch/switch.go

```diff
@@ -894,9 +894,12 @@ func (s *Switch) addLink(link ChannelLink) error {
 		return err
 	}
 
-	log.Infof("Added channel link with chan_id=%v, short_chan_id=(%v), "+
-		"bandwidth=%v", link.ChanID(), spew.Sdump(link.ShortChanID()),
-		link.Bandwidth())
+	// TODO(halseth): remove this after roasbeef's commits.
```

Comment:
After my latest commits within the `htlcswitch`, it should be possible to remove this `go` statement, and not re-introduce the deadlock case.